src/main/java/com/jfrog/ide/idea/scan/utils/ImpactTreeBuilder.java

```diff
@@ -0,0 +1,78 @@
+package com.jfrog.ide.idea.scan.utils;
+
+import com.jfrog.ide.common.nodes.DependencyNode;
+import com.jfrog.ide.common.nodes.DescriptorFileTreeNode;
+import com.jfrog.ide.common.nodes.FileTreeNode;
+import com.jfrog.ide.common.nodes.subentities.ImpactTree;
+import com.jfrog.ide.common.nodes.subentities.ImpactTreeNode;
+
+import java.util.*;
+
+public class ImpactTreeBuilder {
+    public static final int IMPACT_PATHS_LIMIT = 20;
+
+    /**
+     * Builds impact paths for {@link DependencyNode} objects.
+     *
+     * @param vulnerableDependencies a map of component IDs and the DependencyNode object matching each of them
+     * @param parents                a map of all dependencies and their parents
+     * @param rootId                 the project's root component ID
+     * @return list of {@link FileTreeNode}s which are all {@link DescriptorFileTreeNode}s with the given
+     * {@link DependencyNode}s inside them
+     */
+    public static void populateImpactTrees(Map<String, DependencyNode> vulnerableDependencies, Map<String, Set<String>> parents, String rootId) {
```

Comment:
I think it's a bit unclear which of this method's argument is actually an output of the method. Maybe we should explain this better as part of the method description, or better, return the output as a return value.

Reply:
I prefer not to return the DependencyNodes to make it clear that the method doesn't create new DependencyNodes but populates the existing ones. I updated the method description to make it clearer.